rfcs/0181-list-index-syntax.md

```diff
@@ -0,0 +1,290 @@
+---
+feature: list-index-syntax
+start-date: 2024-07-14
+author: rhendric
+co-authors:
+shepherd-team: infinisil
```

Comment:
```suggestion
shepherd-team: infinisil inclyc
```

May I add myself? (Currently maintaining the LSP, so I'd like to subscribe syntax changes!)